Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +911,160 @@ export class OtherSettings {
         this.resource = resource;
     }
 
-    public get editorTabSize(): number | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<number>("tabSize"); }
-    public get editorInsertSpaces(): boolean | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<boolean>("insertSpaces"); }
-    public get editorAutoClosingBrackets(): string | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<string>("autoClosingBrackets"); }
-    public get filesEncoding(): string | undefined { return vscode.workspace.getConfiguration("files", { uri: this.resource, languageId: "cpp" }).get<string>("encoding"); }
-    public get filesAssociations(): any { return vscode.workspace.getConfiguration("files").get("associations"); }
-    public set filesAssociations(value: any) {
-        void vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Workspace);
+    private getAsString(sectionName: string, settingName: string, resource: any, defaultValue: string): string {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        if (section === undefined || section === null) {
+            const message = "Configuration not found";
```

Comment:
Since it's always an error, let's call it 'error' and use "setting" instead of "configuration".
`const error = 'setting does not exist';`